driver/csiplugin/controllerserver.go

```diff
@@ -608,11 +608,166 @@ func (cs *ScaleControllerServer) CreateVolume(ctx context.Context, req *csi.Crea
 	if !allValid {
 		return nil, status.Error(codes.InvalidArgument, "The Parameter(s) not supported in storageClass: "+invalidParams)
 	}
-	scaleVol, err := getScaleVolumeOptions(ctx, req.GetParameters())
+
+	scaleVol, isNewVolumeType, primaryClusterID, err := cs.setScaleVolume(ctx, req, volName, volSize)
+	if err != nil {
+		return nil, err
+	}
+
+	isSnapSource, isVolSource, volSrc, snapIdMembers, srcVolumeIDMembers, err := cs.getVolORSnapMembers(ctx, req, volName)
+	if err != nil {
+		return nil, err
+	}
+
+	err = cs.checkPrimaryFileSetLink(ctx, scaleVol)
```

Comment:
checkPrimaryFileSetLink() could also be used in validateCloneRequest() and validateSnapId() method

Reply:
addressed the comment and made the changes.